frontend/packages/console-dynamic-plugin-sdk/src/api/api.ts

```diff
@@ -30,3 +31,24 @@ export const consoleFetch: ConsoleFetch = newMockImpl();
 export const consoleFetchJSON: ConsoleFetchJSON = newMockImpl();
 mockProperties(consoleFetchJSON, 'delete', 'post', 'put', 'patch');
 export const consoleFetchText: ConsoleFetchText = newMockImpl();
+/**
+ *
+ * @type {import("./component-api-types").NavPage } NavPage
+ *
+ * A component that creates a Navigation bar. It takes array of NavPage objects and renderes a NavBar.
+ * Routing is handled as part of the component.
+ * @example
+ * const HomePage: React.FC = (props) => {
+ *     const page = {
+ *       href: '/home',
+ *       name: 'Home',
+ *       component: () => <>Home</>
+ *     }
+ *     return <HorizontalNav match={props.match} pages={[page]} />
+ * }
+ *
+ * @param {object=} resource - The resource associated with this Navigation, an object of K8sResourceCommon type
+ * @param {NavPage[]} pages - An array of page objects
+ * @param {object} match - match object provided by React Router
+ */
+export const HorizontalNav: React.FC<HorizontalNavProps> = newMockImpl();
```

Comment:
Since a component only has a single param, I don't think having multiple params in jsdoc is correct. You will need to use `typedef` to construct a type for the prop and then assign a single `@param {PropType} props`.

We can always address this later though.

Reply:
I will make changes along with the introduction of a documentation tool. It's a bit difficult to do it without being able to see the exact output.